src/components/admin/users/UserTable.tsx

```diff
@@ -0,0 +1,68 @@
+"use client"
+
+import { type ColumnConfig, DataTable } from "@/components/common/datatable"
+import type { StaffUser } from "@/generated/prisma/client"
+
+export type UserTableProps = {
+  users: StaffUser[]
+  onRowClick?: (user: StaffUser) => void
+}
+
+export const UserTable = ({ users, onRowClick }: UserTableProps) => {
```

Comment:
Perhaps a more generic name so that this table could be used for other data as well